Tabloid/client/src/components/TagEdit.js

```diff
@@ -0,0 +1,58 @@
+import React, { useEffect, useState } from "react";
+import { Button, Form, FormGroup, Label, Input} from "reactstrap";
+import { editTag, getTagById } from "../modules/tagManager";
+import { useHistory, useParams } from "react-router-dom";
+ 
+
+const TagEdit = () => {
+  
+
+  const [tag, setTag] = useState({});
+  const history = useHistory();
+  const {id} = useParams();
+
+
+  useEffect(() => {
+    getTagById(id).then((tag) => setTag(tag));
+  }, [id]);
+
+  const handleInputChange = (evt) => {
+    const value = evt.target.value;
+    const key = evt.target.id;
+
+    const copy = { ...tag };
+
+    copy[key] = value;
+    setTag(copy);
+  };
+
+  const handleSave = (evt) => {
+    evt.preventDefault();
+
+    editTag(tag).then(() => {
+      history.push("/tags");
+    });
+  };
+
+  return (  
+    <Form>
+    <FormGroup>
+      <Label for="name">Name</Label>
+      <Input
+        type="text"
+        name="name"
+        id="name"
+        placeholder={tag.name}
+        value={tag.name}
```

Comment:
This value will be undefined on the first page render. Please make sure this is a controlled component by either setting a name property with the value of an empty string on your initial tag object in state, or including logic here to set the value to an empty string when tag.name is undefined.

Reply:
Fixed